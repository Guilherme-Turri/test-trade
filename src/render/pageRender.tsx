import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};