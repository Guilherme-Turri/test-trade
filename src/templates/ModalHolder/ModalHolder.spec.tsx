import { pageRender } from "../../render/pageRender";
import { fireEvent, screen } from '@testing-library/react'
import { ModalHolder } from ".";

const mockedUseDispatch = jest.fn()
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockedUseDispatch
}));

describe('<ModalHolder>', () => {
  it('should render on screen prperly with Modal Country and close modal', () => {
    const { container } = pageRender(<ModalHolder type='country' />)
    const CountryModal = screen.getByTestId('countriesContainer')
    expect(CountryModal).toBeInTheDocument();
    const closeBtn = screen.getByTestId('Close')
    fireEvent.click(closeBtn)
    expect(mockedUseDispatch).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  })

  it('should render on screen prperly with Modal Season and close modal', () => {
    pageRender(<ModalHolder type='season' />)
    const CountryModal = screen.getByTestId('modalSeasonId')
    expect(CountryModal).toBeInTheDocument();
    const closeBtn = screen.getByTestId('Close')
    fireEvent.click(closeBtn)
    expect(mockedUseDispatch).toHaveBeenCalled();
  })

  it('should render on screen prperly with Modal League and close modal', () => {
    pageRender(<ModalHolder type='league' />)
    const CountryModal = screen.getByTestId('modalLeagueId')
    expect(CountryModal).toBeInTheDocument();
    const closeBtn = screen.getByTestId('Close')
    fireEvent.click(closeBtn)
    expect(mockedUseDispatch).toHaveBeenCalled();
  })

  it('should render on screen prperly with Modal Club and close modal', () => {
    pageRender(<ModalHolder type='club' />)
    const CountryModal = screen.getByTestId('modalClubId')
    expect(CountryModal).toBeInTheDocument();
    const closeBtn = screen.getByTestId('Close')
    fireEvent.click(closeBtn)
    expect(mockedUseDispatch).toHaveBeenCalled();
  })
})