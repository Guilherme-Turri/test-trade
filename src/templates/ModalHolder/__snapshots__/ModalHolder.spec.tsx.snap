// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalHolder> should render on screen prperly with Modal Country and close modal 1`] = `
.c0 {
  position: fixed;
  top: 0px;
  width: 100%;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: white;
}

.c1 {
  position: fixed;
  top: 0;
  right: 0.313rem;
  width: 3.75rem;
  height: 3.75rem;
  cursor: pointer;
  z-index: 10000000;
}

.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  color: #00c555;
  width: 100%;
  height: 100%;
}

.c4 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.938rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.625rem;
  overflow-y: auto;
}

.c5 {
  color: #ff5555;
  font-size: 12px;
}

<div>
  <div
    class="c0"
    data-testid="modalHolderId"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="Close"
      >
        <svg
          aria-hidden="true"
          class="c3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <path
              d="m13.41 12 4.3-4.29a1 1 0 1 0-1.42-1.42L12 10.59l-4.29-4.3a1 1 0 0 0-1.42 1.42l4.3 4.29-4.3 4.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0l4.29-4.3 4.29 4.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
              data-name="close"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="c4"
      data-testid="countriesContainer"
    >
      <p
        class="c5"
      >
        Refaça a busca e/ou verifique as requisições diárias
      </p>
    </div>
  </div>
</div>
`;
